lib/logstash/filters/mutate.rb

```diff
@@ -320,21 +324,32 @@ def convert_string(value)
     value.to_s.force_encoding(Encoding::UTF_8)
   end
 
+  def convert_boolean(value)
+    return true if value =~ TRUE_REGEX
+    return false if value.empty? || value =~ FALSE_REGEX
+    @logger.warn("Failed to convert #{value} into boolean.")
+    value
+  end
+
   def convert_integer(value)
     return 1 if value == true
     return 0 if value == false
-    value.to_i
+    value.squeeze(MUC_CP).tr(MUC_CP, MUC_UP).to_i
```

Comment:
I am not sure about the purpose of `squeeze` here? could you please explain?

Reply:
As discussed on AON, it is to replace multiple consecutive commas and periods. But I take your point that this should probably flag up anomaly between `to_i` and `to_f` handling of multiple `_` separators.